keps/487-kubectl-plugin-listing/README.md

```diff
@@ -0,0 +1,316 @@
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories (Optional)](#user-stories-optional)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Version](#version)
+  - [Describe](#describe)
+  - [List](#list)
+  - [Stop](#stop)
+  - [Test Plan](#test-plan)
+      - [Prerequisite testing updates](#prerequisite-testing-updates)
+    - [Unit Tests](#unit-tests)
+    - [Integration tests](#integration-tests)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+
+Provide an easy way to view and manage kueue abstractions via command line tool.
+
+## Motivation
+
+Create tools that provide useful information in an intuitive way for cluster users and could be later extended as managing tool for cluster administrators.
+As a cluster user I don’t need to dive into the difference between job and Workload concept in kueue and should be able quickly discover the job/Workload and ClusterQueue status or see the error message what is misconfigured/should be fixed.
+
+### Goals
+
+Impement basic kueue CLI commands that was described in PRD:
+
+- Display the Workload status.
+- Describe Workload.
+- List Workloads in specific queue/namespace.
+- List admitted/pending workloads
+- List all ClusterQueues.
+- Describe ClusterQueue.
+- Stop Workload.
+- Stop ClusterQueue.
+
+### Non-Goals
+
+## Proposal
+
+We propose creating a Kubectl plugin that exposes kueue specific information. The main command-line interactions will take the following shape:
+
+```bash
+$kubectl kueue <type> <command> <name> <flags> <options>
+```
+
+Set of supported commands: `list`, `describe`, `stop`.
+Set of supported types: `workload`, `clusterqueue`, `localqueue`.
+Sets of supported flags are different for each command/type.
+
+The plugin could be downloaded directly from the repo or installed by the Krew plugging manager.
+
+### User Stories (Optional)
+
+#### Story 1
+
+As a user submitting workloads, I want to easily see the status of an entire Workload, or check on why my workload is not being admitted. I can do this with the new proposed plugin.
+
+```bash
+# List workloads in the cluster.
+$kubectl kueue workloads get
+# Describe particular workload.
+$kubeclt kueue workload describe <name>
+# Display state and conditions to see what could be wrong with workload.
+$kubeclt kueue workload describe <name> --status
+
+```
+
+#### Story 2
+
+As a cluster admin I want to have an easy way to manage cluster queues:
+
+```bash
+$kubectl kueue clusterqueue stop <queue-name>
+```
+
+### Risks and Mitigations
+
+The API of Workload and ClusterQueue doesn't have a State field. Here we are making assumptions regarding the state of the object based on the values of other fields.
```

Comment:
IIUC, the risk is that the CLI implementation for a commands becomes obsolete due to changes in the Kueue API (for example addition of the dedicated CQ state field). 
Overall, I'm not sure there is a need to mention this risk here as it applies to virtually any feature which uses some of the Kueue API. Also, the mitigation of having a good test coverage seems generic, and applies to any feature. Also, I'm not sure I understand this part "In case of any change the state field should be disabled."

Reply:
Yes you're correct. Yes, it's probably obvious, however I think the risk here is higher than for other features. 
For example in case when we add `stop ClusterQueue` functionality, the State heuristic here should be updated.

> Also, I'm not sure I understand this part "In case of any change the state field should be disabled."

This was added as possible mitigation approach, but I agree it's confusing, so I removed it.

Reply:
I don't have a strong opinion here, but it feels like not specific enough. For example, the "good test coverage" feels applicable to any feature. It does not feel actionable. 

Maybe this part about the "state" field is not really risk & mitigation, but more about "Design details"? 

There we could explain that we compute the `State` field based on other fields. Also, this seems to be the case for the standard `kubectl` command, where for a pod it returns `State` which can be `Terminating` when a pod has `.deletionTimestamp`.

Reply:
Sgtm, done